cnf-certification-test/operator/suite.go

```diff
@@ -318,3 +349,110 @@ func testOperatorSingleCrdOwner(check *checksdb.Check, env *provider.TestEnviron
+			} else {
+				check.LogError("Pod %q is run as root", put.Name)
+				nonCompliantObjects = append(nonCompliantObjects, testhelper.NewPodReportObject(put.Namespace, put.Name, "Pod has been found is run as root", false))
+			}
+		}
+	}
+	check.SetResult(compliantObjects, nonCompliantObjects)
+}
+
+func testOperatorAutomountTokens(check *checksdb.Check, env *provider.TestEnvironment) {
+	var compliantObjects []*testhelper.ReportObject
+	var nonCompliantObjects []*testhelper.ReportObject
+	for i := range env.Operators {
+		operator := env.Operators[i]
+		check.LogInfo("Testing Operator %q", operator)
+		AllPods := findAllPods(env, operator)
+		for _, put := range AllPods {
+			check.LogInfo("Testing Pod %q", put)
+			// Evaluate the pod's automount service tokens and any attached service accounts
+			client := clientsholder.GetClientsHolder()
+			podPassed, newMsg := rbac.EvaluateAutomountTokens(client.K8sClient.CoreV1(), put.Pod)
+			if !podPassed {
+				check.LogInfo("Pod %q have automount service tokens set to false", put)
+				compliantObjects = append(compliantObjects, testhelper.NewPodReportObject(put.Namespace, put.Name, "Pod %q have automount service tokens set to false", true))
+			} else {
+				check.LogError(newMsg)
+				nonCompliantObjects = append(nonCompliantObjects, testhelper.NewPodReportObject(put.Namespace, put.Name, newMsg, false))
+			}
+		}
+	}
+	check.SetResult(compliantObjects, nonCompliantObjects)
+}
+
+func testOperatorReadOnlyFilesystem(check *checksdb.Check, env *provider.TestEnvironment) {
+	var compliantObjects []*testhelper.ReportObject
+	var nonCompliantObjects []*testhelper.ReportObject
+	for i := range env.Operators {
+		operator := env.Operators[i]
+		check.LogInfo("Testing Operator %q", operator)
+		AllPods := findAllPods(env, operator)
+		for _, put := range AllPods {
+			check.LogInfo("Testing Pod %q", put)
+			for _, cut := range put.Containers {
+				check.LogInfo("Testing Container %q", cut.Name)
+				if cut.IsReadOnlyRootFilesystem(check.GetLoggger()) {
+					check.LogInfo("Pod %q container %q is read only root file system.", put.Name, cut.Name)
+					compliantObjects = append(compliantObjects, testhelper.NewPodReportObject(put.Namespace, put.Name, "Pod has been found read only root file system", true))
+				} else {
+					check.LogError("Pod %q container %q is read not only root file system.", put.Name, cut.Name)
+					nonCompliantObjects = append(nonCompliantObjects, testhelper.NewPodReportObject(put.Namespace, put.Name, "Pod has been found read not only root file system", false))
+				}
+			}
+		}
+	}
+	check.SetResult(compliantObjects, nonCompliantObjects)
+}
+
+func findAllPods(env *provider.TestEnvironment, operator *provider.Operator) []*provider.Pod {
+	var AllPods []*provider.Pod
+	for _, put := range env.AllPods {
+		fmt.Printf("pod: %v", put)
+		for _, ownerRef := range put.ObjectMeta.GetOwnerReferences() {
+			for i := range operator.Csv.Spec.CustomResourceDefinitions.Owned {
```

Comment:
This does not look right to me. Is it tested against some operator? 
In [my PR](https://github.com/test-network-function/cnf-certification-test/pull/2130), we get all custom resource pods owned by the operator. So you do not need this logic. My suggestion is to remove this part and test. You can get all Pods from the field mentioned in the PR description.

Reply:
Resolved!